src/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version1_X/MvcViewDocumentClassifierPass.cs

```diff
@@ -25,7 +27,16 @@ public class MvcViewDocumentClassifierPass : DocumentClassifierPassBase
             @namespace.Content = "AspNetCore";
 
             var filePath = codeDocument.Source.RelativePath ?? codeDocument.Source.FilePath;
-            @class.ClassName = CSharpIdentifier.GetClassNameFromPath(filePath);
+            if (string.IsNullOrEmpty(filePath))
```

Comment:
Can we capture some info here about how/why this happens? What's the possible scenarios? The code seems fine

Reply:
You mean add a comment?